packages/web5/lib/src/pexv2/pd.dart

```diff
@@ -0,0 +1,274 @@
+        name: json['name'],
+        purpose: json['purpose'],
+        inputDescriptors: List<InputDescriptor>.from(
+          json['input_descriptors'].map((x) => InputDescriptor.fromJson(x)),
+        ),
+      );
+
+  Map<String, dynamic> toJson() => {
+        'id': id,
+        'name': name,
+        'purpose': purpose,
+        'input_descriptors':
+            List<dynamic>.from(inputDescriptors.map((x) => x.toJson())),
+      };
+
+  List<String> selectCredentials(List<String> vcJwts) {
+    final Set<String> matches = {};
+
+    for (final inputDescriptor in inputDescriptors) {
+      final matchingVcJwts = inputDescriptor.selectCredentials(vcJwts);
+      if (matchingVcJwts.isEmpty) {
+        return [];
+      }
+      matches.addAll(matchingVcJwts);
+    }
+
+    return matches.toList();
+  }
+}
+
+class _TokenizedField {
+  List<String> paths;
+  String token;
+
+  _TokenizedField({required this.paths, required this.token});
+}
+
+/// InputDescriptor represents a DIF Input Descriptor defined [here].
+/// Input Descriptors are used to describe the information a Verifier requires of a Holder.
+///
+/// [here]: https://identity.foundation/presentation-exchange/#input-descriptor
+class InputDescriptor {
+  String id;
+  String? name;
+  String? purpose;
+  Constraints constraints;
+
+  InputDescriptor({
+    required this.id,
+    this.name,
+    this.purpose,
+    required this.constraints,
+  });
+
+  factory InputDescriptor.fromJson(Map<String, dynamic> json) =>
+      InputDescriptor(
+        id: json['id'],
+        name: json['name'],
+        purpose: json['purpose'],
+        constraints: Constraints.fromJson(json['constraints']),
+      );
+
+  Map<String, dynamic> toJson() => {
+        'id': id,
+        'name': name,
+        'purpose': purpose,
+        'constraints': constraints.toJson(),
+      };
+
+  String _generateRandomToken() {
+    final rand = Random.secure();
+    final bytes = Uint8List(16);
+    for (int i = 0; i < 16; i++) {
+      bytes[i] = rand.nextInt(256);
+    }
+    return hex.encode(bytes);
+  }
+
+  List<String> selectCredentials(List<String> vcJWTs) {
+    final List<String> answer = [];
+    final List<_TokenizedField> tokenizedField = [];
+    final schemaMap = {
+      '\$schema': 'http://json-schema.org/draft-07/schema#',
+      'type': 'object',
+      'properties': {},
+      'required': [],
+    };
+
+    // Populate JSON schema and generate tokens for each field
+    for (var field in constraints.fields ?? []) {
+      final token = _generateRandomToken();
+      tokenizedField
+          .add(_TokenizedField(token: token, paths: field.path ?? []));
+
+      final properties = schemaMap['properties'] as Map<String, dynamic>;
+
+      if (field.filter != null) {
+        properties[token] = field.filter.toJson();
+      } else {
+        final anyType = {
+          'type': ['string', 'number', 'boolean', 'object', 'array'],
+        };
+        properties[token] = anyType;
```

Comment:
do we need this? afaict there's no need to include the field in the aggregate json schema if no filter is present

Reply:
fixed